org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/parser/Scanner.java

```diff
@@ -2567,8 +2567,7 @@ public final boolean jumpOverUnicodeWhiteSpace() throws InvalidInputException {
 }
 
 public boolean isInModuleDeclaration() {
-	return this.fakeInModule || this.insideModuleInfo ||
-			(this.activeParser != null ? this.activeParser.isParsingModuleDeclaration() : false);
+	return this.fakeInModule || this.insideModuleInfo;
 }
```

Comment:
I think this change is the most crucial and looks fine. If I look at the write operations to `org.eclipse.jdt.internal.compiler.parser.Scanner.activeParser` there is ONLY one in `org.eclipse.jdt.internal.compiler.parser.Scanner.setActiveParser(ConflictedParser)` - This one already caches whether we are in module declaration.

So the code `this.activeParser != null ? this.activeParser.isParsingModuleDeclaration() : false` is redundant - it is already cached in Scanner#isModuleInfo